### DISCUSSION TITLE
Examples

### DISCUSSION THREAD

**Discussion by ptman** [go-webauthn/webauthn#47] [Q&A] [answered]
Hi!

First of all thanks for taking initiative, forking and cleaning up.

Some questions:

1. Do you have examples of how to use the library somewhere? Preferably something in the repo (that gets compiled so that it doesn't bitrot) or part of the go documentation.
2. Is there a chat somewhere for discussing this library? Gitter? Matrix?
3. There is BeginLogin() and BeginDiscoverableLogin() and ValidateLogin() and ValidateDiscoverableLogin(), but no pair to FinishLogin(). Should there be?

Cheers,
Paul

**Comment by james-d-elliott** [comment, marked as answer]
Hey Paul! Thanks for the inquiry.

1. There are a couple examples listed in the network: https://github.com/go-webauthn/webauthn/network/dependents?package_id=UGFja2FnZS0yOTQ2MjQ2MTE3
    1. The library specific example is [github.com/go-webauthn/example](https://github.com/go-webauthn/example) however this has not been updated in some time. I do plan however to work on a more extensive example which could be used as a building block for a go + react application, including some form of registration (outside of webauthn, and one that leverages webauthn for the username-less flow) as well as adding a CI/CD workflow to both this and the example project, however this is going to be a bit of work, and ideally I'd like to test all functionality from a users perspective even though the ECMAScript portion is not relevant to the library, using the Google Dev tools and go-rod. 
    2. [github.com/authelia/authelia](https://github.com/authelia/authelia) is another project I am a maintainer for which was the motivation for this fork, specifically to address issues that seemed unresolved in the original fork or took too long for my liking to resolve.
2. There is no matrix for this library in specific yet. My personal matrix is in my bio. 
3. FinishLogin was not extended to the discoverable login flow as it's largely not necessary since all the machinery to perform these actions is exported. However I'm not at all opposed to adding it. I had in mind creating a more robust implementation of this in the future once I figured out a signature that could facilitate both without potentially inadvertently encouraging a pattern which could lead to security issues.

I plan to continue to maintain this alongside making PR's to [github.com/duo-labs/go-webauthn](https://github.com/go-webauthn/webauthn). I am currently the only official maintainer, whereas the other is backed by a company. If this ever becomes a popular repo I'll probably look for additional maintainers, and would likely be willing to add anyone eager and knowledgeable upon request. I'm also fairly certain the other Authelia maintainers would get involved upon request.

Please also note that the difference between the repos listed on the README changed slightly due to some of my PR's being merged. I will look to update this shortly.
